### PR TITLE
sonobuoy: update metadata, add update script and version test

### PR DESCRIPTION
**Pull request by katexochen**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/cluster/sonobuoy/update.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+set -euo pipefail
+
+# Do the actual update.
+nix-update "${UPDATE_NIX_ATTR_PATH}"
+
+# Get the src metadata.
+owner=$(nix-instantiate --eval -E "with import ./. {}; ${UPDATE_NIX_ATTR_PATH}.src.owner" | tr -d '"')
+repo=$(nix-instantiate --eval -E "with import ./. {}; ${UPDATE_NIX_ATTR_PATH}.src.repo" | tr -d '"')
+tag=$(nix-instantiate --eval -E "with import ./. {}; ${UPDATE_NIX_ATTR_PATH}.src.rev" | tr -d '"')
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/sonobuoy/update.sh:12]: We can only instantiate this once and output json and parse that

File: pkgs/applications/networking/cluster/sonobuoy/update.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+# Curl the release.
+read -r type tag_sha < <(
+    curl -fsSL "https://api.github.com/repos/${owner}/${repo}/git/ref/tags/${tag}" |
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/sonobuoy/update.sh:16]: This should optional take a GITHUB_TOKEN. See other scripts on how to do this.

File: pkgs/applications/networking/cluster/sonobuoy/update.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    jq -j '.object.type, " ", .object.sha, "\n"'
+)
+
+if [[ "${type}" == "commit" ]]; then
+    sha="${tag_sha}"
+else
+    sha=$(
+        curl -fsSL "https://api.github.com/repos/${owner}/${repo}/git/tags/${tag_sha}" |
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/sonobuoy/update.sh:24]: same here

**Review by SuperSandro2000 — APPROVED** [review]
Didn't try it myself but looks good to me

**Comment by katexochen** [comment]
Rebased to resolve merge conflict after package has been updated on master.

File: pkgs/applications/networking/cluster/sonobuoy/default.nix [hunk 4]
```diff
@@ -27,10 +28,18 @@ buildGoModule rec {
...
+      version = "v${version}";
+    };
+  };
+
   meta = with lib; {
     description = ''
-      Diagnostic tool that makes it easier to understand the
-      state of a Kubernetes cluster.
+      Diagnostic tool that makes it easier to understand the state of a Kubernetes cluster
     '';
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/sonobuoy/default.nix:43]: We could write that to one line to emphasize that it should be short.

**Review by SuperSandro2000 — APPROVED** [review]

**Review by malt3 — APPROVED** [review]

**Comment by katexochen** [comment]
@SuperSandro2000 could you merge this? Thanks!

**Pull request merged**